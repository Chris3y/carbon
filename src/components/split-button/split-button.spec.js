import React from 'react';
import TestRenderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import SplitButtonWithTheme, { SplitButton } from './split-button.component';
import Icon from '../icon';
import Button from '../button';
import { elementsTagTest, rootTagTest } from '../../utils/helpers/tags/tags-specs';
import StyledSplitButton from './split-button.style';
import StyledSplitButtonChildren from './split-button-children.style';
import BaseTheme from '../../style/themes/base';
import classicTheme from '../../style/themes/classic';
import {
  assertStyleMatch,
  keyboard
} from '../../__spec_helper__/test-utils';
import 'jest-styled-components';

const render = (mainProps, childButtons, childProps, renderer = shallow) => {
  return renderer(
    <SplitButton
      { ...mainProps }
      data-element='bar'
      data-role='baz'
    >
      { renderChildButtons(childButtons, childProps) }
    </SplitButton>
  );
};

function renderChildButtons(buttons, childProps) {
  if (buttons.length === 0) {
    const { children } = childProps;
    return (
      <Button { ...childProps }>
        { children }
      </Button>
    );
  }
  return buttons;
}

describe('SplitButton', () => {
  let wrapper, toggle;
  const handleMainButton = jasmine.createSpy('main');
  const handleSecondButton = jasmine.createSpy('second');

  beforeEach(() => {
    wrapper = render(
      {
        text: 'mainButton',
        onClick: handleMainButton
      },
      [],
      {
        onClick: handleSecondButton,
        children: 'Second Button'
      }
    );
    toggle = wrapper.find('[data-element="open"]');
  });

  describe('render with custom className', () => {
    it('sets default state', () => {
      expect(wrapper.instance().state.showAdditionalButtons).toEqual(false);
    });

    it('renders dropdown icon', () => {
      expect(
        toggle.contains(
          <Icon type='dropdown' />
        )
      ).toBeTruthy();
    });
  });

  describe('mouse click dropdown toggle', () => {
    it('prevents default', () => {
      const ev = jasmine.createSpy();
      toggle.simulate('click', { preventDefault: ev });
      expect(ev).toHaveBeenCalled();
    });
  });

  describe('mouse enter dropdown toggle', () => {
    it('changes showAdditionalButtons state', () => {
      toggle.simulate('mouseenter');
      expect(wrapper.state().showAdditionalButtons).toEqual(true);
    });

    it('renders additional buttons to match snapshot', () => {
      toggle.simulate('mouseenter');
      expect(wrapper).toMatchSnapshot();
    });

    it('when disabled it does not change the state', () => {
      const wrapper2 = render(
        { text: 'mainButton', disabled: true }, [], { children: 'Second Button' }
      );
      const block = wrapper2.find('[data-element="open"]');
      block.simulate('mouseenter');
      expect(wrapper2.state().showAdditionalButtons).toEqual(false);
    });
  });

  describe('mouse leave split-button', () => {
    let wrapper2, toggle2, mainButton;
    beforeEach(() => {
      wrapper2 = render({ text: 'mainButton' }, [], { children: 'Second Button' });
      mainButton = wrapper2.find('[data-element="main-button"]');
      toggle2 = wrapper2.find('[data-element="open"]');
    });

    it('changes showAdditionalButtons state', () => {
      toggle2.simulate('mouseenter');
      expect(wrapper2.state().showAdditionalButtons).toEqual(true);
      mainButton.simulate('mouseenter');
      wrapper2.instance().forceUpdate();
      expect(wrapper2.state().showAdditionalButtons).toEqual(false);
    });

    it('hides additional buttons', () => {
      toggle2.simulate('mouseenter');
      expect(wrapper2.containsMatchingElement(
        <Button>Second Button</Button>
      )).toBeTruthy();

      mainButton.simulate('mouseenter');
      expect(wrapper2.containsMatchingElement(
        <Button>Second Button</Button>
      )).toBeFalsy();
    });
  });

  describe('clicking a button', () => {
    it('the handler should be called on the main button', () => {
      toggle.simulate('mouseenter');
      wrapper.instance().forceUpdate();
      const button = wrapper.find('[data-element="additional-buttons"]').find(Button);
      button.simulate('click');
      expect(handleSecondButton).toHaveBeenCalled();
    });
  });

  describe('keyboard accessibility of additional buttons', async () => {
    describe('pressing the "up" key', () => {
      let wrapper2;
      beforeEach(() => {
        wrapper2 = render({
          text: 'mainButton',
          'data-element': 'bar',
          'data-role': 'baz'
        },
        [
          <Button>Extra Button</Button>,
          <Button>Extra Button</Button>,
          <Button>Extra Button</Button>
        ],
        {}, mount);
        toggle = wrapper2.find('[data-element="open"]').hostNodes();
        toggle.prop('onFocus')();
      });
      it('adds focus to the button referenced by the index', () => {
        const { additionalButtons } = wrapper2.instance();
        for (let index = additionalButtons.length - 1; index >= 0; index--) {
          const button = additionalButtons[index];
          const spy = spyOn(button, 'focus');
          keyboard.pressUpArrow();
          expect(spy).toBeCalled();
        }
        // setInterval(() => spys.forEach(spy => expect(spy).toHaveBeenCalled(), 200));
      });

      fit('matches the expected style for the button indexed', () => {
        const { additionalButtons } = wrapper2.instance();
        for (let index = additionalButtons.length - 1; index >= 0; index--) {
          keyboard.pressUpArrow();
          const button = additionalButtons[index];
          assertStyleMatch({
            background: 'transparent',
            borderColor: BaseTheme.colors.primary,
            color: BaseTheme.colors.primary,
            fontSize: '14px',
            height: '40px',
            paddingLeft: '24px',
            paddingRight: '24px'
          }, button);
        }
      });

      afterEach(() => {
        wrapper2.unmount();
      });
    });

    describe('pressing the "down" key', () => {
      let wrapper3;
      beforeEach(() => {
        wrapper3 = render({
          text: 'mainButton',
          'data-element': 'bar',
          'data-role': 'baz'
        },
        [
          <Button>Extra Button</Button>,
          <Button>Extra Button</Button>,
          <Button>Extra Button</Button>
        ],
        {}, mount);
        toggle = wrapper3.find('[data-element="open"]').hostNodes();
        toggle.prop('onFocus')();
      });
      // it('adds focus to the button referenced by the index', () => {
      //   toggle = wrapper2.find('[data-element="open"]').hostNodes();
      //   const { additionalButtons } = wrapper2.instance();
      //   toggle.prop('onFocus')();
      //   wrapper2.props().children.forEach((_, index) => {
      //     keyboard.pressDownArrow();
      //     const button = additionalButtons[index];
      //     const spy = spyOn(button, 'focus');
      //     setInterval(() => expect(spy).toHaveBeenCalled(), 200);
      //   });
      // });

      fit('matches the expected style for the button indexed', () => {
        const { additionalButtons } = wrapper3.instance();
        // console.log('split', wrapper3.instance().additionalButtons);
        additionalButtons.forEach((_, index) => {
          const button = additionalButtons[index];
          keyboard.pressDownArrow();
          // console.log('test focus call');
          assertStyleMatch({
            background: 'transparent',
            borderColor: BaseTheme.colors.primary,
            color: BaseTheme.colors.primary,
            fontSize: '14px',
            height: '40px',
            paddingLeft: '24px',
            paddingRight: '24px'
          }, button);
        });
      });
      afterEach(() => {
        wrapper3.unmount();
      });
    });
  });

  describe('button refs', () => {
    beforeEach(() => {
      wrapper = render({
        text: 'mainButton',
        'data-element': 'bar',
        'data-role': 'baz'
      },
      [
        <Button>Extra Button</Button>,
        <Button>Extra Button</Button>
      ],
      {}, mount);
    });

    it('creates and stores refs for the main and toggle buttons', () => {
      const { splitButtons } = wrapper.instance();
      expect(splitButtons.length).toEqual(2);
      splitButtons.forEach(btn => expect(btn).not.toEqual(null));
      expect(splitButtons[0]).not.toEqual(wrapper.instance().splitButtons[1]);
    });

    it('creates and stores refs for the any additonal buttons', () => {
      toggle = wrapper.find('[data-element="open"]').hostNodes();
      toggle.prop('onFocus')();
      const { additionalButtons } = wrapper.instance();
      expect(additionalButtons.length).toEqual(2);
      additionalButtons.forEach(btn => expect(btn).not.toEqual(null));
      expect(additionalButtons[0]).not.toEqual(wrapper.instance().splitButtons[1]);
    });

    afterEach(() => {
      wrapper.unmount();
    });
  });

  describe('tags', () => {
    describe('on component', () => {
      it('include correct component, element and role data tags', () => {
        rootTagTest(wrapper, 'split-button', 'bar', 'baz');
      });
    });

    describe('on internal elements', () => {
      wrapper = shallow(
        <SplitButton text='Test'>
          <Button>Test1</Button>
          <Button>Test2</Button>
        </SplitButton>
      );
      wrapper.setState({ showAdditionalButtons: true });

      elementsTagTest(wrapper, [
        'additional-buttons',
        'main-button',
        'open'
      ]);
    });
  });
});
