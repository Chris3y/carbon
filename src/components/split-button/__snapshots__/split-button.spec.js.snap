// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledSplitButton for business themes renders styles correctly 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border: 2px solid transparent;
  box-sizing: border-box;
  font-weight: 600;
  padding-top: 1px;
  padding-bottom: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: transparent;
  border-color: #26A826;
  color: #26A826;
  font-size: 14px;
  height: 40px;
  padding-left: 24px;
  padding-right: 24px;
}

.c2:focus {
  outline: solid 3px #FFB500;
}

.c2 + .c1 {
  margin-left: 16px;
}

.c2:hover {
  background: #006300;
  border-color: #006300;
  color: #FFFFFF;
}

.c2 .carbon-icon {
  margin-left: 0px;
  margin-right: 8px;
}

.c5 + .c1 {
  margin-left: 16px;
}

.c6 + .c1 {
  margin-left: 16px;
}

.c7 + .c1 {
  margin-left: 16px;
}

.c8 + .c1 {
  margin-left: 16px;
}

.c9 + .c1 {
  margin-left: 16px;
}

.c0 {
  display: inline-block;
  position: relative;
}

.c0 .c1 {
  margin: 0;
}

.c0 .c1:focus {
  outline-offset: -3px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border: 2px solid transparent;
  box-sizing: border-box;
  font-weight: 600;
  padding-top: 1px;
  padding-bottom: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: transparent;
  border-color: #26A826;
  color: #26A826;
  font-size: 14px;
  height: 40px;
  padding-left: 24px;
  padding-right: 24px;
  border-left: none;
  padding: 0 10px;
}

.c4:focus {
  outline: solid 3px #FFB500;
}

.c4 + .c3 {
  margin-left: 16px;
}

.c4:hover {
  background: #006300;
  border-color: #006300;
  color: #FFFFFF;
}

.c4 .carbon-icon {
  margin-left: 8px;
  margin-right: 0px;
}

.c1 + .c4 {
  margin-left: 0;
  border-left: none;
}

.c1 + .c4:focus {
  margin-left: -2px;
  padding-left: 12px;
}

.c1 + .c4 .carbon-icon {
  margin-left: 0;
}

.c10 + .c3 {
  margin-left: 16px;
}

.c1 + .c10 {
  margin-left: 0;
  border-left: none;
}

.c1 + .c10:focus {
  margin-left: -2px;
  padding-left: 12px;
}

.c1 + .c10 .carbon-icon {
  margin-left: 0;
}

.c11 + .c3 {
  margin-left: 16px;
}

.c1 + .c11 {
  margin-left: 0;
  border-left: none;
}

.c1 + .c11:focus {
  margin-left: -2px;
  padding-left: 12px;
}

.c1 + .c11 .carbon-icon {
  margin-left: 0;
}

.c12 .c1 {
  background-color: #006300;
  border: 1px solid #006300;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c12 .c1:focus {
  background-color: #004400;
}

.c12 .c1:hover {
  background-color: #004400;
}

.c12 .c1 + .c17 .c1 {
  margin-top: 3px;
}

.c13 .c1 {
  background-color: #006045;
  border: 1px solid #006045;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c13 .c1:focus {
  background-color: #003F2E;
}

.c13 .c1:hover {
  background-color: #003F2E;
}

.c13 .c1 + .c17 .c1 {
  margin-top: 3px;
}

.c13 .c1 {
  background-color: #1e499f;
  border: 1px solid #1e499f;
  color: #FFF;
  font-size: 14px;
  height: 31px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 0 18px;
  text-align: left;
}

.c13 .c1:focus {
  background-color: #163777;
}

.c13 .c1:hover {
  background-color: #163777;
}

.c14 .c1 {
  background-color: #006045;
  border: 1px solid #006045;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c14 .c1:focus {
  background-color: #003F2E;
}

.c14 .c1:hover {
  background-color: #003F2E;
}

.c14 .c1 + .c17 .c1 {
  margin-top: 3px;
}

.c15 .c1 {
  background-color: #005B9A;
  border: 1px solid #005B9A;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c15 .c1:focus {
  background-color: #004372;
}

.c15 .c1:hover {
  background-color: #004372;
}

.c15 .c1 + .c17 .c1 {
  margin-top: 3px;
}

.c16 .c1 {
  background-color: #4F2775;
  border: 1px solid #4F2775;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c16 .c1:focus {
  background-color: #3D1E5B;
}

.c16 .c1:hover {
  background-color: #3D1E5B;
}

.c16 .c1 + .c17 .c1 {
  margin-top: 3px;
}

<div
  aria-haspopup="true"
  className="c0"
  data-component="split-button"
  onMouseLeave={[Function]}
>
  <div>
    <button
      className="c1 c2"
      data-component="button"
      data-element="main-button"
      disabled={false}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      role="button"
      size="medium"
    >
      <span>
        <span
          data-element="main-text"
        >
          Mock text
        </span>
      </span>
    </button>
    <button
      aria-controls="guid-12345"
      aria-expanded={false}
      aria-haspopup="true"
      className="c1 c3 c4"
      data-element="open"
      disabled={false}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      onMouseEnter={[Function]}
      size="medium"
    >
      <span
        className="carbon-icon icon-dropdown"
        data-component="icon"
        data-element="dropdown"
      />
    </button>
  </div>
</div>
`;

exports[`StyledSplitButton for the classic theme renders styles correctly 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border: 2px solid transparent;
  box-sizing: border-box;
  font-weight: 600;
  padding-top: 1px;
  padding-bottom: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: transparent;
  border-color: #26A826;
  color: #26A826;
  font-size: 14px;
  height: 40px;
  padding-left: 24px;
  padding-right: 24px;
}

.c2:focus {
  outline: solid 3px #FFB500;
}

.c2 + .c1 {
  margin-left: 16px;
}

.c2:hover {
  background: #006300;
  border-color: #006300;
  color: #FFFFFF;
}

.c2 .carbon-icon {
  margin-left: 0px;
  margin-right: 8px;
}

.c5 + .c1 {
  margin-left: 16px;
}

.c6 + .c1 {
  margin-left: 16px;
}

.c7 + .c1 {
  margin-left: 16px;
}

.c8 + .c1 {
  margin-left: 16px;
}

.c9 + .c1 {
  margin-left: 16px;
}

.c0 {
  display: inline-block;
  position: relative;
}

.c0 .c1 {
  margin: 0;
}

.c0 .c1:focus {
  outline-offset: -3px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  border: 2px solid transparent;
  box-sizing: border-box;
  font-weight: 600;
  padding-top: 1px;
  padding-bottom: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: transparent;
  border-color: #26A826;
  color: #26A826;
  font-size: 14px;
  height: 40px;
  padding-left: 24px;
  padding-right: 24px;
  border-left: none;
  padding: 0 10px;
}

.c4:focus {
  outline: solid 3px #FFB500;
}

.c4 + .c3 {
  margin-left: 16px;
}

.c4:hover {
  background: #006300;
  border-color: #006300;
  color: #FFFFFF;
}

.c4 .carbon-icon {
  margin-left: 8px;
  margin-right: 0px;
}

.c1 + .c4 {
  margin-left: 0;
  border-left: none;
}

.c1 + .c4:focus {
  margin-left: -2px;
  padding-left: 12px;
}

.c1 + .c4 .carbon-icon {
  margin-left: 0;
}

.c10 + .c3 {
  margin-left: 16px;
}

.c1 + .c10 {
  margin-left: 0;
  border-left: none;
}

.c1 + .c10:focus {
  margin-left: -2px;
  padding-left: 12px;
}

.c1 + .c10 .carbon-icon {
  margin-left: 0;
}

.c11 + .c3 {
  margin-left: 16px;
}

.c1 + .c11 {
  margin-left: 0;
  border-left: none;
}

.c1 + .c11:focus {
  margin-left: -2px;
  padding-left: 12px;
}

.c1 + .c11 .carbon-icon {
  margin-left: 0;
}

.c12 .c1 {
  background-color: #006300;
  border: 1px solid #006300;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c12 .c1:focus {
  background-color: #004400;
}

.c12 .c1:hover {
  background-color: #004400;
}

.c12 .c1 + .c17 .c1 {
  margin-top: 3px;
}

.c13 .c1 {
  background-color: #006045;
  border: 1px solid #006045;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c13 .c1:focus {
  background-color: #003F2E;
}

.c13 .c1:hover {
  background-color: #003F2E;
}

.c13 .c1 + .c17 .c1 {
  margin-top: 3px;
}

.c13 .c1 {
  background-color: #1e499f;
  border: 1px solid #1e499f;
  color: #FFF;
  font-size: 14px;
  height: 31px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 0 18px;
  text-align: left;
}

.c13 .c1:focus {
  background-color: #163777;
}

.c13 .c1:hover {
  background-color: #163777;
}

.c14 .c1 {
  background-color: #006045;
  border: 1px solid #006045;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c14 .c1:focus {
  background-color: #003F2E;
}

.c14 .c1:hover {
  background-color: #003F2E;
}

.c14 .c1 + .c17 .c1 {
  margin-top: 3px;
}

.c15 .c1 {
  background-color: #005B9A;
  border: 1px solid #005B9A;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c15 .c1:focus {
  background-color: #004372;
}

.c15 .c1:hover {
  background-color: #004372;
}

.c15 .c1 + .c17 .c1 {
  margin-top: 3px;
}

.c16 .c1 {
  background-color: #4F2775;
  border: 1px solid #4F2775;
  color: #FFFFFF;
  display: block;
  margin-left: 0;
  margin-top: 3px;
  margin-bottom: 3px;
  min-width: 100%;
  z-index: 10;
}

.c16 .c1:focus {
  background-color: #3D1E5B;
}

.c16 .c1:hover {
  background-color: #3D1E5B;
}

.c16 .c1 + .c17 .c1 {
  margin-top: 3px;
}

<div
  aria-haspopup="true"
  className="c0"
  data-component="split-button"
  onMouseLeave={[Function]}
>
  <div>
    <button
      className="c1 c2"
      data-component="button"
      data-element="main-button"
      disabled={false}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      role="button"
      size="medium"
    >
      <span>
        <span
          data-element="main-text"
        >
          Mock text
        </span>
      </span>
    </button>
    <button
      aria-controls="guid-12345"
      aria-expanded={false}
      aria-haspopup="true"
      className="c1 c3 c4"
      data-element="open"
      disabled={false}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      onMouseEnter={[Function]}
      size="medium"
    >
      <span
        className="carbon-icon icon-dropdown"
        data-component="icon"
        data-element="dropdown"
      />
    </button>
  </div>
</div>
`;
